tree/ntuple/v7/inc/ROOT/RPageAllocator.hxx

```diff
@@ -77,6 +79,56 @@ public:
    static void DeletePage(const RPage &page);
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::Detail::RPageAllocatorCache
+\ingroup NTuple
+\brief Reuse memory allocations from a small thread-local cache
+
+This allocator returns pages that have at least the required capacity. `DeletePage()` does not immediately deallocate
+memory; instead, pages are returned to a thread-local cache, dropping the smallest allocated buffer if the cache is full.
+If a previous page cannot be recycled, `AllocT` is used to allocate memory.
+*/
+// clang-format on
+template <typename AllocT = RPageAllocatorHeap, std::size_t CacheSize = 16>
+class RPageAllocatorCache {
+public:
+   thread_local static struct RPageCache {
+      std::deque<RPage> fPages;
```

Comment:
Does `fPages` have to be public?

Reply:
Misplaced `public:`; thanks for spotting!